pkgs/http/lib/src/base_response.dart

```diff
@@ -68,3 +70,68 @@ abstract class BaseResponse {
     }
   }
 }
+
+// "token" as defined in RFC 2616, 2.2
+// See https://datatracker.ietf.org/doc/html/rfc2616#section-2.2
+const _tokenChars = r"!#$%&'*+\-.0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ^_`"
+    'abcdefghijklmnopqrstuvwxyz|~';
+
+// Splits comma-seperated header values.
+var _headerSplitter = RegExp(r'[ \t]*,[ \t]*');
+
+// Splits comma-seperated "Set-Cookie" header values.
+//
+// Set-Cookie strings can contain commas. In particular, the following
+// productions defined in RFC-6265, section 4.1.1:
+// - <sane-cookie-date> e.g. "Expires=Sun, 06 Nov 1994 08:49:37 GMT"
+// - <path-value> e.g. "Path=somepath,"
+// - <extension-av> e.g. "AnyString,Really,"
+//
+// Some values are ambiguous e.g.
+// "Set-Cookie: lang=en; Path=/foo/"
+// "Set-Cookie: SID=x23"
+// and:
+// "Set-Cookie: lang=en; Path=/foo/,SID=x23"
+// would both be result in `response.headers` => "lang=en; Path=/foo/,SID=x23"
+//
+// The idea behind this regex is that ",<valid token>=" is more likely to
+// start a new <cookie-pair> then be part of <path-value> or <extension-av>.
+//
+// See https://datatracker.ietf.org/doc/html/rfc6265#section-4.1.1
+var _setCookieSplitter = RegExp(r'[ \t]*,[ \t]*(?=[' + _tokenChars + r']+=)');
+
+extension HeadersWithFieldLists on BaseResponse {
+  /// The HTTP headers returned by the server.
+  ///
+  /// The header names are converted to lowercase and stored with their
+  /// associated header values.
+  ///
+  /// Cookies can be parsed using the dart:io `Cookie` class:
+  ///
+  /// ```dart
+  /// import "dart:io";
+  /// import "package:http/http.dart";
+  ///
+  /// void main() async {
+  /// final response = await Client().get(Uri.https('example.com', '/'));
+  /// final cookies = [
+  ///   for (var value i
+  ///       in response.headersFieldValueList['set-cookie'] ?? <String>[])
+  ///     Cookie.fromSetCookieValue(value)
+  /// ];
+  Map<String, List<String>> get headersFieldValueList {
```

Comment:
This name is pretty long - can we consider any shorter ones? Maybe `headerSplitValues`?

Reply:
Done